### ISSUE TITLE
Middleman + External pipeline (webpack) not adding assets to sitemap on linux

### ISSUE THREAD

**Issue by mbajur** [middleman/middleman#2118]
## Expected behavior and actual behavior

I have a middleman instance created with a following template: https://gitlab.com/nemuco-org/middleman-webpack

It works perfectly fine when i'm running it on OSX/macOS. `middleman s` runs webpack and middleman and watches for file changes. Generated bundles are added to the sitemap.

However, when i'm trying to run the same app, middleman seems to not be cathing up files generated by webpack. I have tried it on several instances of linux (in virtualbox, on remote virtual machine, on virtualbox on another PC, on ubuntu in docker and on alpine in docker) and it acts exactly the same way.  Webpack generates the bundle files but middleman is not adding them to the sitemap for some reason. 

Not sure if that's important but first noticable difference is that, on my macOS, last output after running `middleman s` is

```
== LiveReloading path: /assets/javascript/bootstrap.bundle.js
== LiveReloading path: /assets/javascript/bootstrapcss.bundle.js
== LiveReloading path: /assets/javascript/main.bundle.js
== LiveReloading path: /assets/stylesheets/bootstrapcss.bundle.css
== LiveReloading path: /assets/stylesheets/main.bundle.css
```

while there is not such logs on linux.

## Steps to reproduce the problem (from a clean middleman installation)

I've prepared a simple docker setup to make it easier to debug it:

    $ middleman init middleman_webpack_bug -T git@gitlab.com:nemuco-org/middleman-webpack.git
    $ cd middleman_webpack_bug
    $ make setup && make start

Open lcalhost:4567 and see that styles are not loaded. 

## Additional information

- Ruby version: 2.3.3
- Middleman version: 4.2.1
- OS version: macOS Sierra / Ubuntu >= 14.04

**Comment by vinc** [comment]
Same problem on Arch Linux with ruby `2.4.2` and middleman `4.2.1`.

    $ git clone https://github.com/grassdog/middleman-webpack
    $ bundle install
    $ yarn install
    $ middleman server

I expect to be able to get `/assets/javascript/main.bundle.js` but the server return a 404 error.

The file `.tmp/dist/assets/javascript/main.bundle.js` exists, and I can do:

    $ middleman build
    $ cd build
    $ python -m http.server

To get it (with the same name if I deactivate `:asset_hash` in `config.rb`).

**Comment by carlthuringer** [comment]
I'm also experiencing this issue running Middleman in a Docker container.

**Comment by tdreyno** [comment]
@carlthuringer Are you able to share a Docker image to test in?

**Comment by evenreven** [comment]
I'm having this issue too. I'm on Arch with rvm-installed Ruby 2.4.3 and Middleman 4.2.1. `middleman build` works, `middleman server` does not. To rule out environment issues, I've also tried the same on Ubuntu through a Virtualbox Vagrant container. Same result. 

It would make sense if middleman server starts before the yarn/webpack step and that's why the files are missing, but livereload should take care of this, no? Or could it be that after the sitemap is generated, it won't accept assets from other directories?

I know a little Ruby, but I'm basically green when it comes to webpack and js, and I have no idea how to troubleshoot this.

**Comment by DavideSilva** [comment]
@mbajur try removing the clean-webpack-plugin from your webpack.config.js. Middleman is watching the .tmp folder and by deleting it, it causes Middleman to serve the wrong files.

**Comment by evenreven** [comment]
I commented out all references to clean-webpack-plugin, removed yarn.lock and node_modules and ran yarn again, and it looks like it's working! But why does it work on MacOS, I wonder? Any idea?

**Comment by DavideSilva** [comment]
What we managed to find was that Middleman starts watching the correct files but when the clean-webpack-plugin runs and deletes the .tmp/ folder, Middleman stops pointing to the output folder and instead searches the source, causing problems. In MacOS this doesn't seem to be an issue but on Linux it is.

The files_for_sitemap method in middleman-core/lib/middleman-core/sitemap/extensions/on_disk.rb is where we detected that change. On startup, the (source) watchers contain the correct files but after the clean-webpack-plugin run, the watchers stop watching the output folder and revert to the source folder.
